discretize/mixins/mpl_mod.py

```diff
@@ -0,0 +1,2070 @@
+        >>> h3 = np.linspace(.1, .5, 3)
+        >>> mesh = discretize.TensorMesh([h1, h2, h3])
+        >>> mesh.plot_grid(nodes=True, faces=True, centers=True, lines=True)
+        >>> plt.show()
+
+        Plotting a 2D CurvilinearMesh
+
+        >>> from matplotlib import pyplot as plt
+        >>> import discretize
+        >>> X, Y = discretize.utils.exampleLrmGrid([10, 10], 'rotate')
+        >>> M = discretize.CurvilinearMesh([X, Y])
+        >>> M.plot_grid()
+        >>> plt.show()
+
+        Plotting a 3D CurvilinearMesh
+
+        >>> from matplotlib import pyplot as plt
+        >>> import discretize
+        >>> X, Y, Z = discretize.utils.exampleLrmGrid([5, 5, 5], 'rotate')
+        >>> M = discretize.CurvilinearMesh([X, Y, Z])
+        >>> M.plot_grid()
+        >>> plt.show()
+
+        Plotting a 2D TreeMesh
+
+        >>> from matplotlib import pyplot as plt
+        >>> import discretize
+        >>> M = discretize.TreeMesh([32, 32])
+        >>> M.insert_cells([[0.25, 0.25]], [4])
+        >>> M.plot_grid()
+        >>> plt.show()
+
+        Plotting a 3D TreeMesh
+
+        >>> from matplotlib import pyplot as plt
+        >>> import discretize
+        >>> M = discretize.TreeMesh([32, 32, 32])
+        >>> M.insert_cells([[0.3, 0.75, 0.22]], [4])
+        >>> M.plot_grid()
+        >>> plt.show()
+        """
+        mesh_type = self._meshType.lower()
+        plotters = {
+            'tree' : self.__plot_grid_tree,
+            'tensor': self.__plot_grid_tensor,
+            'curv': self.__plot_grid_curv,
+            'cyl': self.__plot_grid_cyl,
+        }
+        try:
+            plotter = plotters[mesh_type]
+        except KeyError:
+            raise NotImplementedError('Mesh type `{}` does not have a plot_grid implementation.'.format(type(self).__name__))
+
+
+        if "showIt" in kwargs:
+            show_it = kwargs.pop("showIt")
+            warnings.warn("showIt has been deprecated, please use show_it", FutureWarning)
+
+        if ax is not None:
+            ax_test = ax
+            if not isinstance(ax, (list, tuple, np.ndarray)):
+                ax_test = (ax, )
+            for a in ax_test:
+                if not isinstance(a, matplotlib.axes.Axes):
+                    raise TypeError("ax must be an matplotlib.axes.Axes")
+        elif mesh_type != 'cyl':
+            axOpts = {'projection': '3d'} if self.dim == 3 else {}
+            plt.figure()
+            ax = plt.subplot(111, **axOpts)
+
+        rcParams = rc_params()
+        if lines:
+            kwargs['color'] = kwargs.get('color', rcParams['lines.color'])
+            kwargs['linewidth'] = kwargs.get('linewidth', rcParams['lines.linewidth'])
+
+        return plotter(
+            ax=ax, nodes=nodes, faces=faces, centers=centers, edges=edges,
+            lines=lines, show_it=show_it, **kwargs
+        )
+
+    def plot_image(
+        self, v, v_type='CC', grid=False, view='real',
+        ax=None, clim=None, show_it=False,
```

Comment:
I would vote to deprecate `clim` everywhere, for a couple of reasons.
- matplotlib uses `vmin`/`vmax`, so it can be confusing
- It should really be provided in `pcolor_opts{'vmin': -1e-10, 'vmax': 1e10}`.
- If a norm is used, then it has to be provided in the norm, e.g., `pcolor_opts{'norm': LogNorm(vmin=1e-10, vmax=1e-5)}`

There are a couple of PR's (#226, #228) and issues (#218) related to this, so this big overhaul might be the right moment to deprecate `clim` entirely, IMHO.